packages/commons/src/events/Action.ts

```diff
@@ -62,18 +61,28 @@ export const ActionInputBase = z.object({
   )
 })
 
-export const ActionInput = z.union([
-  ActionInputBase.extend({
-    type: z.enum([ActionType.CREATE])
-  }),
-  ActionInputBase.extend({
-    type: z.enum([ActionType.REGISTER]),
-    identifiers: z.object({
-      trackingId: z.string(),
-      registrationNumber: z.string()
-    })
+export const CreateActionInput = ActionInputBase.extend({})
+export const NotifyActionInput = ActionInputBase.extend({})
```

Comment:
Could you add a comment for this pattern? Is it a matter of taking copy of the base an separating the `type` extension?

Reply:
`RegisterActionInput.extend({ type: z.enum([ActionType.REGISTER]) })` since one could argue that this is not `RegisterActionInput` until it has the type